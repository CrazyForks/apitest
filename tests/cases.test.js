const { spwanTest } = require("./utils");

describe("cases", () => {
  test("invalid prop key", async () => {
    const { stdout, code } = await spwanTest("cases/invalid-prop-key.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid unit value type", async () => {
    const { stdout, code } = await spwanTest("cases/invalid-unit-value-type.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid group value type", async () => {
    const { stdout, code } = await spwanTest("cases/invalid-unit-value-type.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("default-describe", async () => {
    const { stdout, code } = await spwanTest("cases/default-describe.jsona", ["--reset"]);
    expect(code).toEqual(0);
    expect(stdout).toMatchSnapshot();
  });
  test("merge mixin", async () => {
    const { stdout, code } = await spwanTest("cases/merge-mixin.jsona", ["--reset"]);
    expect(code).toEqual(0);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid mixin value type", async () => {
    const { stdout, code } = await spwanTest("cases/invalid-mixin-type.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid unit mixin", async () => {
    const { stdout, code } = await spwanTest("cases/invalid-unit-mixin.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid unit mixin type", async () => {
    const { stdout, code } = await spwanTest("cases/invalid-unit-mixin-type.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid unit client type", async () => {
    const { stdout, code } = await spwanTest("cases/invalid-unit-client-type.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("client", async () => {
    const { code } = await spwanTest("cases/client.jsona", ["--reset"]);
    expect(code).toEqual(0);
  });
});
