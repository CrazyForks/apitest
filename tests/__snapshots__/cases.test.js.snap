// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cases default-describe 1`] = `
"[1mmodule main[22m
[1m[22m[1m  group group1[22m
[1m[22m[1m    unit unit1[22m[32m âœ”[39m
[32m[39m"
`;

exports[`cases invalid group value type 1`] = `
"main.test1: should have object value at line 2 col 3
"
`;

exports[`cases invalid mixin value type 1`] = `
"main.test1@mixin: req1 should be object at line 3 col 29
"
`;

exports[`cases invalid prop key 1`] = `
"main: prop 'test a' should satify rules of variable name at line 2 col 3
"
`;

exports[`cases invalid unit client type 1`] = `
"main.test1@client: should have string or object value at line 2 col 13
"
`;

exports[`cases invalid unit mixin 1`] = `
"main.test1@mixin: reqx is miss at line 3 col 29
"
`;

exports[`cases invalid unit mixin type 1`] = `
"main.test1@mixin: should have string or array value at line 3 col 29
"
`;

exports[`cases invalid unit value type 1`] = `
"main.test1: should have object value at line 2 col 3
"
`;

exports[`cases merge mixin 1`] = `
"[1mmodule main[22m
[1m[22m[1m  single mixin[22m[32m âœ”[39m
[32m[39m[1m  multiple mixin[22m[32m âœ”[39m
[32m[39m[1m  omit mixin if exist prop[22m[32m âœ”[39m
[32m[39m[1m  nest mixin[22m[32m âœ”[39m
[32m[39m"
`;
